packages/asset-swapper/src/utils/market_operation_utils/pools_cache/sgPoolDataService.ts

```diff
@@ -0,0 +1,114 @@
+import { ChainId } from '@0x/contract-addresses';
+import { logUtils } from '@0x/utils';
+import { PoolDataService, SubgraphPoolBase } from '@balancer-labs/sdk';
+import { gql, request } from 'graphql-request';
+
+const queryWithLinear = gql`
+    query fetchTopPoolsWithLinear($maxPoolsFetched: Int!) {
+        pools: pools(
+            first: $maxPoolsFetched
+            where: { swapEnabled: true }
+            orderBy: totalLiquidity
+            orderDirection: desc
+        ) {
+            id
+            address
+            poolType
+            swapFee
+            totalShares
+            tokens {
+                address
+                balance
+                decimals
+                weight
+                priceRate
+            }
+            tokensList
+            totalWeight
+            amp
+            expiryTime
+            unitSeconds
+            principalToken
+            baseToken
+            swapEnabled
+            wrappedIndex
+            mainIndex
+            lowerTarget
+            upperTarget
+        }
+    }
+`;
+
+const queryWithOutLinear = gql`
+    query fetchTopPoolsWithoutLinear($maxPoolsFetched: Int!) {
+        pools: pools(
+            first: $maxPoolsFetched
+            where: { swapEnabled: true }
+            orderBy: totalLiquidity
+            orderDirection: desc
+        ) {
+            id
+            address
+            poolType
+            swapFee
+            totalShares
+            tokens {
+                address
+                balance
+                decimals
+                weight
+                priceRate
+            }
+            tokensList
+            totalWeight
+            amp
+            expiryTime
+            unitSeconds
+            principalToken
+            baseToken
+            swapEnabled
+        }
+    }
+`;
+
+const QUERY_BY_CHAIN_ID: { [chainId: number]: string } = {
+    [ChainId.Mainnet]: queryWithLinear,
+    [ChainId.Polygon]: queryWithOutLinear,
+};
+
+const DEFAULT_MAX_POOLS_FETCHED = 96;
```

Comment:
This number really impacts how long AS takes to boot up so don't set it too high.